### PR TITLE
openssl-util: add common implementation of digest+sign

### PR DESCRIPTION
**Pull request by poettering**

### CODE REVIEW

**Comment by poettering** [comment]
No new code, just unification of two existing very similar pieces of code

**Comment by poettering** [comment]
force pushed a new version that hopefully addresses the issue the CI found.

**Review by bluca — APPROVED** [review]

File: src/shared/openssl-util.c [hunk 1]
```diff
@@ -545,6 +545,49 @@ int pubkey_fingerprint(EVP_PKEY *pk, const EVP_MD *md, void **ret, size_t *ret_s
...
+                if (size == SIZE_MAX) /* If SIZE_MAX input is a string whose size we determine automatically */
+                        size = strlen(data);
+        }
+
+        _cleanup_(EVP_MD_CTX_freep) EVP_MD_CTX* mdctx = EVP_MD_CTX_new();
+        if (!mdctx)
+                return log_oom_debug();
+
+        if (EVP_DigestSignInit(mdctx, NULL, md, NULL, privkey) != 1)
+                return log_debug_errno(SYNTHETIC_ERRNO(ENOTRECOVERABLE), "Failed to initialize signature context.");
```
Comment (daandemeyer) [inline src/shared/openssl-util.c]: We have this fancy `log_openssl_errors()` macro now that we should probably use here

**Review by DaanDeMeyer — CHANGES_REQUESTED** [review]

File: src/shared/openssl-util.c (hunk 1 above)
Comment (poettering) [reply src/shared/openssl-util.c]: addressed

**Pull request merged**